integrations/criteo/lib/index.js

```diff
@@ -212,7 +221,7 @@ Criteo.prototype.setExtraData = function() {
   // Criteo has a special tag for emails.
   // They also require all emails get passed as md5 hashes.
   if (traits.email) {
-    ret.push({ event: 'setEmail', email: md5(traits.email) });
+    ret.push({ event: 'setHashedEmail', email: md5(traits.email) });
```

Comment:
I thought we talked about calling `setCustomerId` *only* if `setHashedEmail` is not called. Right now, if a customer sends `email` and `userId`, we make both calls.

Reply:
Got it now.

Reply:
Also Regarding productId, i changed the sequence like:
1) Trim the productId if its type is string, so that string with only space will take care first.
2) then Checking for **!productId**